packages/2d/src/components/Node.ts

```diff
@@ -908,6 +908,163 @@ export class Node implements Promisable<Node> {
     }
   }
 
+  /**
+   * Find all descendants of this node that match the given predicate.
+   *
+   * @param predicate - A function that returns true if the node matches.
+   */
+  public find<T extends Node>(predicate: (node: any) => node is T): T[];
+  /**
+   * Find all descendants of this node that match the given predicate.
+   *
+   * @param predicate - A function that returns true if the node matches.
+   */
+  public find<T extends Node = Node>(predicate: (node: any) => boolean): T[];
+  public find<T extends Node>(predicate: (node: any) => node is T): T[] {
```

Comment:
`find` on the JS array returns one element—the first one. Shouldn't this be `filter`.

Reply:
Changed it to `findAll` in relation to the discord discussion.